spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -661,6 +664,41 @@ public int getUnconfirmedCount() {
 		}
 	}
 
+	@Override
+	public void start() {
+	}
+
+	@Override
+	public void stop() {
+		synchronized (this.directReplyToContainers) {
+			if (this.directReplyToContainers.size() > 0) {
+				Iterator<Entry<ConnectionFactory, DirectReplyToMessageListenerContainer>> iterator =
+						this.directReplyToContainers.entrySet().iterator();
+				while (iterator.hasNext()) {
+					Entry<ConnectionFactory, DirectReplyToMessageListenerContainer> entry = iterator.next();
+					if (entry.getValue().isRunning()) {
+						entry.getValue().stop();
+					}
+					iterator.remove();
```

Comment:
Why `this.directReplyToContainers.clear()` isn't good in the end instead ?
OTOH:
```
	@Override
	public void stop() {
		synchronized (this.directReplyToContainers) {
				this.directReplyToContainers.values()
						.stream()
						.filter(AbstractMessageListenerContainer::isRunning)
						.forEach(AbstractMessageListenerContainer::stop);

				this.directReplyToContainers.clear();
		}
	}
```